cusub_cortex/state_machine/src/tasks/start_gate_det.py

```diff
@@ -0,0 +1,685 @@
+                    if self.count == 1:
+                        self.in_position_state = self.drive_client.get_standard_state()
+                    if self.in_position_state is not None:
+                        self.drive_client.set_setpoint(self.in_position_state, loop=False)
+                    self.count += 1
+                    # Stop Sub
+                    if self.count > self.count_target:
+                        self.cuprint("centered")
+                        self.yaw_client.set_setpoint(0, loop=False)
+                        self.drive_client.increment_setpoint(0.15, loop=False)
+                        rospy.sleep(5)
+                        break
+                elif az_diff > 0.3:
+                    self.cuprint("az_diff: " + str(az_diff))
+                    # Strafe
+                    if self.orbit_right:
+                        strafe_msg.data = strafe_set + self.strafe_carrot
+                    else:
+                        strafe_msg.data = strafe_set - self.strafe_carrot
+                    self.strafe_pub.publish(strafe_msg)
+                else:
+                    # Not within threshold
+                    self.count = 0
+
+                self.listener.clear_new_dv_flag(dobj_num)
+
+                    
+            if userdata.timeout_obj.timed_out:
+                self.yaw_client.disable()
+                self.drive_client.disable()
+                self.depth_client.disable()
+                userdata.outcome = "timed_out"
+                return "not_found"
+                
+            self.depth_client.set_setpoint(-0.8, loop=False)
+            r.sleep()
+
+        self.yaw_client.disable()
+        self.drive_client.disable()
+        self.depth_client.disable()
+        return "centered"
+
+
+    def eval_startgate_poles(self, dobj_nums):
+        [az_0, el_0, mag_0] = self.listener.get_avg_bearing(dobj_nums[0], num_dv=5)
+        [az_1, el_1, mag_1] = self.listener.get_avg_bearing(dobj_nums[1], num_dv=5)
+        [az_2, el_2, mag_2] = self.listener.get_avg_bearing(dobj_nums[2], num_dv=5)
+
+        if az_0 > az_1 > az_2:
+            self.pole_l = dobj_nums[0]
+            self.pole_r = dobj_nums[2]
+            self.pole_c = dobj_nums[1]
+
+        elif az_2 > az_1 > az_0:
+            self.pole_l = dobj_nums[2]
+            self.pole_r = dobj_nums[0]
+            self.pole_c = dobj_nums[1]
+
+        elif az_0 > az_2 > az_1:
+            self.pole_l = dobj_nums[0]
+            self.pole_r = dobj_nums[1]
+            self.pole_c = dobj_nums[2]
+
+        elif az_1 > az_2 > az_0:
+            self.pole_l = dobj_nums[2]
+            self.pole_r = dobj_nums[0]
+            self.pole_c = dobj_nums[2]
+
+        elif az_1 > az_0 > az_2:
+            self.pole_l = dobj_nums[1]
+            self.pole_r = dobj_nums[2]
+            self.pole_c = dobj_nums[0]
+
+        else:
+            self.pole_l = dobj_nums[2]
+            self.pole_r = dobj_nums[1]
+            self.pole_c = dobj_nums[0]
```

Comment:
Nice, that's some pole decision logic

Reply:
Its very Lukesian, you could do this in 1 line with sort and lambda, its the luke way though I suppose.

Reply:
looks like there's an error in the 4th elif statement, missing a dobj 1 determination

Reply:
Mike I'm going to need your code to stick to Lukesian programming standards